controller/src/controller.rs

```diff
@@ -296,6 +327,35 @@ fn is_initial_state(brs: &BottlerocketShadow) -> bool {
         Some(status) => status.current_state == BottlerocketShadowState::default(),
     }
 }
+
+fn within_time_window() -> Result<bool> {
+    let time_window_start_env = env::var(PREVENT_UPDATE_TIME_WINDOW_START_ENV_VAR).context(
+        controllerclient_error::MissingEnvVariable {
+            variable: PREVENT_UPDATE_TIME_WINDOW_START_ENV_VAR.to_string(),
+        },
+    )?;
+
+    let time_window_end_env = env::var(PREVENT_UPDATE_TIME_WINDOW_END_ENV_VAR).context(
+        controllerclient_error::MissingEnvVariable {
+            variable: PREVENT_UPDATE_TIME_WINDOW_END_ENV_VAR.to_string(),
+        },
+    )?;
+
+    let current_time = Utc::now().time();
+    let time_window_start = NaiveTime::parse_from_str(&time_window_start_env, "%H:%M:%S").context(
+        controllerclient_error::ConvertToNativeTime {
+            date: time_window_start_env,
+        },
+    )?;
+    let time_window_end = NaiveTime::parse_from_str(&time_window_end_env, "%H:%M:%S").context(
+        controllerclient_error::ConvertToNativeTime {
+            date: time_window_end_env,
+        },
+    )?;
+
+    return Ok((current_time > time_window_start) && (current_time < time_window_end));
```

Comment:
I have two concerns here:
1. What if user accidentally put start time later than end time? 
2. What if user want to block 11pm - 2 am (next day) slot?

Reply:
That's correct concern! I realized it as well and I'm working on the solution.

If users accidentally put start time later than end time, I'll assume this situation as `start time to end time (next day)`. For example, as you mentioned start time `11pm` is later than `2 am`, so brupop will recognize it as 11pm - 2 am (next day) slot. 

I can't recognize `if user accidentally put start time later than end time` as an invalid input since we have valid input like cross a day.